### PR TITLE
Translation framework

### PR DESCRIPTION
**Pull request by biwano**
Integrates a translation framework (linguijs)
Notes:
- The framework is only integrated in the app, not the site.
- This is a very minimal effort (It does not use nextjs internationalization standards, It does not autodetect the user language, It does not save the user preference, The menu to choose the language is ugly)

### CODE REVIEW

**Comment by 0xAurelius** [comment]
Resolves #29

File: app/pages/_app.tsx [hunk 1]
```diff
@@ -2,9 +2,25 @@ import "@klimadao/lib/theme/colors.css";
...
+import { I18nProvider } from "@lingui/react";
+import { i18n } from "@lingui/core";
+
+import { init } from "../lib/i18n";
 
 function MyApp({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />;
+  useEffect(() => {
+    // Activate the default locale on page load
+    init();
```
Comment (Atmosfearful) [inline app/pages/_app.tsx:15]: Does the fact that this only runs after mount cause a flash of un-translated text?

File: app/package.json [hunk 2]
```diff
@@ -2,13 +2,21 @@
...
+    "lingui:extract": "NODE_ENV=test lingui extract",
+    "lingui:compile": "NODE_ENV=test lingui compile",
+    "postinstall" : "npm run lingui:compile"
   },
   "dependencies": {
+    "@babel/core": "^7.16.0",
+    "@lingui/cli": "3.0.3",
+    "@lingui/macro": "3.0.3",
+    "@lingui/react": "3.0.3",
+    "typescript": "4.0.5",
```
Comment (Atmosfearful) [inline app/package.json:19]: `typescript` and `@lingui/cli` should be dev-dependencies

I believe `@lingui/macro` should be as well

File: app/package.json [hunk 3]
```diff
@@ -2,13 +2,21 @@
...
   "private": true,
   "scripts": {
-    "dev": "next dev -p 3001",
+    "dev": "npm run lingui:compile && next dev -p 3001",
     "build": "next build",
     "start": "next start",
     "export": "next export",
-    "lint": "next lint"
+    "lint": "next lint",
+    "lingui:extract": "NODE_ENV=test lingui extract",
```
Comment (Atmosfearful) [inline app/package.json:10]: why `test`?

File: app/locale/en/messages.po [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+msgid "footer.docs"
+msgstr "docs"
+
+#: components/views/Home/index.tsx:323
+msgid "footer.home"
+msgstr "home"
+
+#: components/views/Home/index.tsx:235
+msgid "header.connectedto"
+msgstr "You are connected to"
```
Comment (Atmosfearful) [inline app/locale/en/messages.po:34]: I could probably answer this question by reading the docs but,
does this mean in order to change a string, we need to change it in the component, but also here in the messages file?

Or is this auto-generated?

File: app/components/views/Home/index.tsx [hunk 5]
```diff
@@ -268,7 +274,7 @@ export const Home: FC = () => {
     <nav className={styles.nav}>
       {chainId === 80001 && (
         <p className={styles.testnet_warning}>
-          ⚠️You are connected to <strong>testnet</strong>
+          ⚠️<Trans id="header.connectedto">You are connected to </Trans><strong>testnet</strong>
```
Comment (Atmosfearful) [inline app/components/views/Home/index.tsx:277]: This is probably fine for now but it raised a question for me:
Generally speaking, is it better to use interpolation for strings like these?
What does that look like in lingui? in other libs I used its something like:
```json
"header.connectedto": "You are connected to {networkName}"
```

Comment (Atmosfearful) [reply app/components/views/Home/index.tsx:277]: the rationale being, in another language the noun `testnet` might not appear at the end

File: app/package.json [hunk 6]
```diff
@@ -2,13 +2,21 @@
...
+    "lingui:compile": "NODE_ENV=test lingui compile",
+    "postinstall" : "npm run lingui:compile"
   },
   "dependencies": {
+    "@babel/core": "^7.16.0",
```
Comment (Atmosfearful) [inline app/package.json:15]: are you sure we need babel, doesn't lingui and nextjs have it as a subdependency already?

**Review by Atmosfearful — CHANGES_REQUESTED** [review]
I'm super stoked for this, insanely helpful. I added a few comments and questions, and there are two more issues before I can merge to the i18n branch:

1. (read my comments below before doing this) It looks like you generated some package-locks. There should only be one, at the root, which is a feature of `npm workspaces`. Try deleting the extras, removing the new dependencies from package.json, and reinstalling using the workspace command, from the root klimadao folder:
`npm install --save-dev -w app -w site @lingui/macro @lingui/cli`
and
`npm install -w app -w site @lingui/react`

That should install them, symlink them to the workspaces (so each is only installed once for all packages 🔥 💯 ), and add them to the root package-lock.json.

2. Please rebase on the latest staging and reinstall deps, then run `pretter --write` to format your code. With the new setup, builds will fail if not formatted.

Thank you!!

File: app/components/views/Home/index.tsx (hunk 5 above)
Comment (kargakis) [reply app/components/views/Home/index.tsx:277]: If we can't use interpolation in translations I would say this should be a deal-breaker for lingui.

File: app/pages/_app.tsx (hunk 1 above)
Comment (biwano) [reply app/pages/_app.tsx:15]: I think so. I am a react/nextjs noob so I do not really know the lifecycles etc... Let me know if there is a better place to put init stuff.

File: app/package.json (hunk 3 above)
Comment (biwano) [reply app/package.json:10]: There seem to be a bug with `lingui extract` when running it in production mode in particular setups. This is a workaround
https://github.com/lingui/js-lingui/issues/433

File: app/locale/en/messages.po (hunk 4 above)
Comment (biwano) [reply app/locale/en/messages.po:84]: You need to change it only in the `messages.po` files. The strings in the component are only default values.

File: app/components/views/Home/index.tsx (hunk 5 above)
Comment (biwano) [reply app/components/views/Home/index.tsx:277]: Interpolation is definetly better and possible with lingui. I was just being lazy.

File: app/package.json (hunk 2 above)
Comment (biwano) [reply app/package.json:19]: I like to compile translations as part of the build process. But i am okay with commiting js files too.

File: app/package.json (hunk 6 above)
Comment (biwano) [reply app/package.json:15]: Good catch. I was following installation instructions blindly.

**Comment by biwano** [comment]
I found a problem with the t component when translating the stake page. When trying this like the documentation says:
`t({
   id: "msg.refresh",
   message: "Refresh inbox"
})`
The compiler complains about the signature of t. 

I had to use a mix of:
- `defineMessage` (so lingui:extract picks up the translation)
- `i18n` (to do the actual translation)

File: app/pages/_app.tsx (hunk 1 above)
Comment (Atmosfearful) [reply app/pages/_app.tsx:15]: OK we can fix this in a follow-up PR :)

**Comment by Atmosfearful** [comment]
As mentioned in my DM please upgrade the package-lock and then this can merge to the i18n branch.

**Review by Atmosfearful — APPROVED** [review]
Nice work, let's merge to the i18n branch, work out the remaining issues/questions, and hopefully get this to production ASAP! 🚢 🚢

**Comment by Atmosfearful** [comment]
Reminder: We will need to do regular rebases of staging to the i18n branch to keep things up to date

**Pull request merged**